cloudbridge/cloud/providers/aws/services.py

```diff
@@ -842,11 +842,46 @@ def instance_data(self):
         return [vm_type for vm_type in vm_types_list
                 if vm_type.get('pricing', {}).get(self.provider.region_name)]
 
+    @dispatch(event="provider.compute.vm_types.get",
+              priority=BaseVMTypeService.STANDARD_EVENT_PRIORITY)
+    def get(self, vm_type_id):
+        vm_type = (AWSVMType(self.provider, vm_type)
+                   for vm_type in self.instance_data
+                   if vm_type['instance_type'] == vm_type_id)
+        return next(vm_type, None)
+
+    @cachetools.cached(cachetools.TTLCache(maxsize=1, ttl=24 * 3600))
+    def _list_vm_types_in_zone(self, zone_name):
+        result = (self.provider.ec2_conn.meta.client
+                      .describe_reserved_instances_offerings(
+                          AvailabilityZone=zone_name))
+        ids = list(set(offering.get('InstanceType') for offering
+                       in result.get('ReservedInstancesOfferings')))
+        while result.get('NextToken'):
+            result = (self.provider.ec2_conn.meta.client
+                          .describe_reserved_instances_offerings(
```

Comment:
I cannot find anywhere in the AWS docs if this is the case. In the best scenario, that's the case. In the worse scenario, it's better than what we have at the moment, particularly because evidently a select zone for one account does not need to map to the same zone for another account:

>An Availability Zone is represented by a region code followed by a letter identifier; for example, us-east-1a. To ensure that resources are distributed across the Availability Zones for a region, we independently map Availability Zones to names for each AWS account. For example, the Availability Zone us-east-1a for your AWS account might not be the same location as us-east-1a for another AWS account.

See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-regions-availability-zones

Even if does represent the current availability, it cannot account for the dynamic nature of the system (eg, someone requests the last available instance between this method returning and a request for an instance is issued). If it does turn out to be current availability, we can just bump the caching to a lower value...